inventory-service/src/main/java/com/spaghettininjas/yaposs/controller/ProductController.java

```diff
@@ -0,0 +1,63 @@
+package com.spaghettininjas.yaposs.controller;
+
+import com.spaghettininjas.yaposs.repository.entity.Product;
+import com.spaghettininjas.yaposs.service.ProductService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/api/Products")
+public class ProductController {
+
+    private final ProductService service;
+
+    public ProductController(ProductService service) {
+        this.service = service;
+    }
+
+    @GetMapping(path = "/{id}")
+    public ResponseEntity<Product> getById(@PathVariable int id) {
+        return service.findById((long) id)
+                .map(inventory -> ResponseEntity.ok().body(inventory))
+                .orElse(ResponseEntity.notFound().build());
+    }
+
+    @DeleteMapping(path = "/{id}")
+    public void deleteById(@PathVariable int id){
+        service.deleteById((long) id);
+    }
+
+    @GetMapping
+    public ResponseEntity<Iterable<Product>> findAll(
+            @RequestParam(required = false, defaultValue = "0") Integer page,
+            @RequestParam(required = false, defaultValue = "10") Integer pageSize
+            )
+    {
+        Iterable<Product> products = service.findAll(page, pageSize);
+        return new ResponseEntity<>(products, HttpStatus.OK);
+    }
+
+    @PutMapping(path = "/{id}")
+    public ResponseEntity<Product> updateProduct(
+            @PathVariable int id,
+            @RequestBody Product product
+    ) {
+        return service.findById((long)id)
+                .map(p -> {
+                    p.setPrice(product.getPrice());
+                    p.setName(product.getName());
+                    p.setCategory(product.getCategory());
+                    return ResponseEntity.status(HttpStatus.OK).body(service.save(p));
+                })
+                .orElseGet(() -> ResponseEntity.notFound().build());
+
+    }
+
+    @PostMapping
+    public ResponseEntity<Product> addProduct(@RequestBody Product product) {
+        return ResponseEntity.status(HttpStatus.CREATED).body(service.save(product));
```

Comment:
This kinda almost exactly like `PUT`, since `repository.save()` overwrites entities with the same id, except the status code might be considered wrong (being pedantic with status codes here 😄  ):

> If a new resource is created, the origin server MUST inform the user agent
>   via the 201 (Created) response. If an existing resource is modified,
>   either the 200 (OK) or 204 (No Content) response codes SHOULD be sent
>   to indicate successful completion of the request.
>
> https://www.rfc-editor.org/rfc/rfc2616#section-9.6